src/object/cli_shard.c

```diff
@@ -121,14 +121,14 @@ dc_rw_cb_singv_lo_get(daos_iod_t *iods, d_sg_list_t *sgls, uint32_t iod_nr,
 	singv_los = reasb_req->orr_singv_los;
 	for (i = 0; i < iod_nr; i++) {
 		singv_lo = &singv_los[i];
-		if (singv_lo->cs_even_dist == 0 || singv_lo->cs_bytes != 0)
+		iod = &iods[i];
+		sgl = &sgls[i];
+		if (singv_lo->cs_even_dist == 0 || singv_lo->cs_bytes != 0 ||
+		    iod->iod_size == DAOS_REC_ANY)
```

Comment:
I thought if ion_size = ANY, then it will not fetch any data? no?

Reply:
for fetch singv, if iod_size=ANY it just means user does not know the actual size and will firstly query from server, original code will trigger the below assertion if the queried iod_size is really zero, so just remove the assertion.